pkg/minikube/registry/drvs/hyperkit/hyperkit.go

```diff
@@ -85,5 +92,70 @@ func status() registry.State {
 		return registry.State{Installed: true, Error: fmt.Errorf("%s failed:\n%s", strings.Join(cmd.Args, " "), out), Fix: "Run 'brew install hyperkit'", Doc: docURL}
 	}
 
+	// Split version from v0.YYYYMMDD-HH-xxxxxxx or 0.YYYYMMDD to YYYYMMDD
+	currentVersion := convertVersionToDate(string(out))
+	specificVersion := splitHyperKitVersion(getHyperKitVersion())
+	// If current hyperkit is not newer than minimumVersion, suggest upgrade information
+	isNew, err := isNewerVersion(currentVersion, specificVersion)
+	if err != nil {
+		return registry.State{Installed: true, Healthy: true, Error: fmt.Errorf("hyperkit version check failed:\n%s", err), Doc: docURL}
+	}
+	if !isNew {
+		return registry.State{Installed: true, Healthy: true, Error: fmt.Errorf("the installed hyperkit version (0.%s) is older than the minimum recommended version (%s)", currentVersion, getHyperKitVersion()), Fix: "Run 'brew upgrade hyperkit'", Doc: docURL}
+	}
+
 	return registry.State{Installed: true, Healthy: true}
 }
+
+// isNewerVersion checks whether current hyperkit is newer than specific version
+func isNewerVersion(currentVersion string, specificVersion string) (bool, error) {
+	// Convert hyperkit version to time.Date to compare whether hyperkit version is not old.
+	layout := "20060102"
+	currentVersionDate, err := time.Parse(layout, currentVersion)
+	if err != nil {
+		glog.Warningf("Unable to parse to time.Date: %v", err)
```

Comment:
In general we either: return errors, or log them. We almost never do  both, as it often causes duplicate errors to be logged.

In order to get the same basic effect here, use:

`return false, errors.Wrap(err, "parse date")`

Reply:
> In general we either: return errors, or log them. We almost never do both, as it often causes duplicate errors to be logged.

Thanks. 
I'll remember this!